### ISSUE TITLE
Subtitle editing

### ISSUE THREAD

**Issue by mifi** [mifi/lossless-cut#250]
suggested function

**Comment by mifi** [comment]
Seems like HTML5 has built in support for subtitles. Just need to convert it to their format.
https://developer.mozilla.org/en-US/docs/Web/Guide/Audio_and_video_delivery/Adding_captions_and_subtitles_to_HTML5_video
https://iandevlin.com/html5/track/index-multiple.html

**Comment by gtknowledgeshare** [comment]
I was looking for "Transcript" support especially for [Microsoft Stream's VTT files](https://web.microsoftstream.com/VTTCleaner/CleanVTT.html) that one can download for auto generated transcripts/subtitles.

[Any update on this](https://github.com/mifi/lossless-cut/issues/418#issuecomment-657552887)? Despite lossy this would be huge step forward for hearing disabilities where burning in the transcript/subtitles to host on a site/player that doesn't support displaying them via codec/video file (or for convenience).

**Comment by mifi** [comment]
Can vtt be converted to srt and vice versa using ffmpeg? If so, then I believe it would be just another subtitle format that would automatically be supported if we add support for subtitle editing in losslesscut.

**Comment by gtknowledgeshare** [comment]
Yes VTT can be converted to SRT like https://subtitletools.com/convert-to-srt-online and sure editing them to correct mistakes or timing could be beneficial for subtitles in general I guess.

But I'm referring to burning them into the video visually itself, even if that means lossy it would make the process so much easier than using a video editing program to place subtitles over the video and re-encode. 
I haven't found any other app that does this.

**Comment by mifi** [comment]
Ah, you want to burn in subtlties. you're right, that is not a lossless operation, and will need a total re-encode of the video. I know this can be done with ffmpeg. I find it odd if there is no GUI programs for doing this though. I will add it to the list of lossy operations feature requests here: #372

> Btw. I just tried dragging in both .srt and .mp4 and when selecting first the mp4 and then srt to drag it in then the .srt is on the top position which causes the merged output to be saved as .srt instead of ,mp4

Sorry it's a bit confusing. merging actually means time-domain, so it's like a concat, which can obviously not be done with a subtitle and a video. In order to add a subtitle track to an existing mp4, you first drag drop ONLY the mp4. then when it's loaded, then drag drop a subtitle.

**Comment by gtknowledgeshare** [comment]
Gotcha. Perhaps the logic for dropping in both at once could be improved to handle this i.e. separate tick box to treat subtitles to be loaded into the video (when only two files are dropped in) . =)

I tried it but didn't work. I can open a separate issue but here fyi

`Error: Command failed with exit code 1: C:\Users\Markus~1\AppData\Local\Temp\1lgrToPkxih26i5yzUeF1I4J5eW\resources\node_modules\ffmpeg-ffprobe-static\ffmpeg.exe -hide_banner -i C:\Users\markus\Videos\video editing\transcript\Forget the Stigma.mp4 -i C:\Users\markus\Videos\video editing\transcript\[SubtitleTools.com] Forget the Stigma_AutoGeneratedCaption.srt -c copy -map 0:0 -map 0:1 -map 1:0 -map_metadata 0 -movflags +faststart -ignore_unknown -f mp4 -y C:\Users\markus\Videos\video editing\transcript\Forget the Stigma-00.00.00.000-00.01.34.314-seg1.mp4`

And attached the vtt and converted srt transcript created for the video https://www.youtube.com/watch?v=ClJc3jRsJNE&t=7s
[transcript.zip](https://github.com/mifi/lossless-cut/files/6867453/transcript.zip)

**Comment by mifi** [comment]
@gtknowledgeshare I tried to youtube-dl your file as an mkv, then download your zip and add the srt to the mkv as I described above, and for me it works. I think mp4 doesn't support srt, but mkv does. Possibly see also #418

**Comment by gtknowledgeshare** [comment]
sorry @mifi  should have thought about that lol thanks for checking.

**Comment by mifi** [comment]
basic subtitle viewing added:

<img width="703" alt="Screenshot 2021-08-28 at 14 27 35" src="https://user-images.githubusercontent.com/402547/131210245-3f844680-1c24-4811-8a43-70a39ce3b8b2.png">

**Comment by taozi1996** [comment]
Is it possible to load local subtitle from pc for viewing like youtube cc subtitle?

**Comment by mifi** [comment]
> Is it possible to load local subtitle from pc for viewing like youtube cc subtitle?

not yet. a workaround is to add the subtitle to a video file, export it, then reload the exported file and view the subtitle

**Comment by gjaldb** [comment]
Hi, a feature request:

Would it be possible to have an option to include the subtitle overlay on frame grabs?

(and my reason for not just taking a manual screenshot or using other software is the convenient naming of the frame grabs in lossless cut with filename and timestamp)

**Comment by mifi** [comment]
It’ possible but it’s not super trivial so I’m not sure it will be implemented unless more people also want that

**Comment by erdemdev** [comment]
> Is it possible to load local subtitle from pc for viewing like youtube cc subtitle?

I really need this feature so I can ditch VLC player 😄

**Comment by appleflyerv3** [comment]
hello please add this functionality. it would be a very good feature.

**Comment by mifi** [comment]
You can already add a .srt file to a video. Then export and re-import

**Comment by this-is-just-con-tribute** [comment]
> You can already add a .srt file to a video. Then export and re-import

Yeah but I edited the subtitles on the full video before editing it because I figured it would make it easier to tell where I am in the video, but then realized that lossless cut would be a better tool for this job before I actually, you know, EDIT it, so that's why I want this feature so I can edit the video and the subtitles don't need to be burned in. But it's not like the video is ruined with the subtitles being burned in, but I just don't wanna do that. But yeah, I can always just burn them in, that works.

**Comment by MoYan-Zhang** [comment]
When importing subtitle files into MP4 files, is there a function to edit the subtitle track language? Currently, the exported subtitle track language displays as “und”.

**Comment by mifi** [comment]
> When importing subtitle files into MP4 files, is there a function to edit the subtitle track language? Currently, the exported subtitle track language displays as “und”.

Did you try to go into the tracks panel and choose the subtitle track "edit track metadata" and then edit or add a Tag named `language`?

**Comment by MoYan-Zhang** [comment]
I have tried to edit metadata. I don't know if there is something wrong with my operation. The track name has not changed. It is still the same "und".



---Original---
From: "Mikael ***@***.***&gt;
Date: Thu, May 28, 2026 22:08 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [mifi/lossless-cut] Subtitle viewing/editing (#250)


mifi left a comment (mifi/lossless-cut#250)
  
When importing subtitle files into MP4 files, is there a function to edit the subtitle track language? Currently, the exported subtitle track language displays as “und”.
  
Did you try to go into the tracks panel and choose the subtitle track "edit track metadata" and then edit or add a Tag named language?
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications, keep track of coding agent tasks and review pull requests on the go with GitHub Mobile for iOS and Android. Download it today!

**Comment by mifi** [comment]
did you use the exact name `language` (all lower case) for the metadata tag and then use the three letter value, e.g. `eng`?

**Comment by MoYan-Zhang** [comment]
I successfully changed the metadata tag "language" to "eng"; what should "Chinese" be named?



---Original---
From: "Mikael ***@***.***&gt;
Date: Thu, May 28, 2026 22:20 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [mifi/lossless-cut] Subtitle viewing/editing (#250)


mifi left a comment (mifi/lossless-cut#250)
 
did you use the exact name language (all lower case) for the metadata tag and then use the three letter value, e.g. eng?
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications, keep track of coding agent tasks and review pull requests on the go with GitHub Mobile for iOS and Android. Download it today!

**Comment by MoYan-Zhang** [comment]
I succeeded! Originally, I used the two letters "zh," but now I've successfully transformed it into "Chinese" using the three-letter "chi" in the orbital language. Thank you so much!!!



---Original---
From: "Mikael ***@***.***&gt;
Date: Thu, May 28, 2026 22:20 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [mifi/lossless-cut] Subtitle viewing/editing (#250)


mifi left a comment (mifi/lossless-cut#250)
 
did you use the exact name language (all lower case) for the metadata tag and then use the three letter value, e.g. eng?
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications, keep track of coding agent tasks and review pull requests on the go with GitHub Mobile for iOS and Android. Download it today!

**Comment by mifi** [comment]
awesome. I will improve this by showing an info icon next to "language"
